archeryutils/tests/test_targets.py

```diff
@@ -236,3 +238,103 @@ def test_min_score_invalid_face_type(self) -> None:
             # Silence mypy as scoring_system must be a valid literal ScoringSystem
             target.scoring_system = "InvalidScoringSystem"  # type: ignore[assignment]
             target.min_score()
+
+    @pytest.mark.parametrize(
+        "scoring_system, diam",
+        [
+            ("5_zone", 122),
+            ("10_zone", 80),
+            ("WA_field", 80),
+            ("IFAA_field", 50),
+            ("Beiter_hit_miss", 6),
+            ("Worcester", (16, "inch")),
+            ("10_zone_6_ring", 80),
+            ("10_zone_5_ring_compound", 40),
+        ],
+    )
+    def test_get_face_spec(self, scoring_system, diam) -> None:
+        """Check that target returns face specs from supported scoring systems."""
+        expected_spec = {
+            "5_zone": {0.244: 9, 0.488: 7, 0.732: 5, 0.976: 3, 1.22: 1},
+            "10_zone": {
+                0.08: 10,
+                0.16: 9,
+                0.24: 8,
+                0.32: 7,
+                0.4: 6,
+                0.48: 5,
+                0.56: 4,
+                0.64: 3,
+                0.72: 2,
+                0.8: 1,
+            },
+            "WA_field": {0.08: 6, 0.16: 5, 0.32: 4, 0.48: 3, 0.64: 2, 0.8: 1},
+            "IFAA_field": {0.1: 5, 0.3: 4, 0.5: 3},
+            "Beiter_hit_miss": {0.06: 1},
+            "Worcester": {0.08128: 5, 0.16256: 4, 0.24384: 3, 0.32512: 2, 0.4064: 1},
+            "10_zone_6_ring": {0.08: 10, 0.16: 9, 0.24: 8, 0.32: 7, 0.4: 6, 0.48: 5},
+            "10_zone_5_ring_compound": {0.02: 10, 0.08: 9, 0.12: 8, 0.16: 7, 0.2: 6},
+        }
```

Comment:
So they totally could be in there and were before... its just that with the extra indentation of being in there, the formatter was vommiting them all over the place, it was almost impossible to read. Possibly with ruff perfering to force to full multi lines it might be better now, lets see:

```python
    @pytest.mark.parametrize(
        "scoring_system, diam, expected_spec",
        [
            ("5_zone", 122, {0.244: 9, 0.488: 7, 0.732: 5, 0.976: 3, 1.22: 1}),
            (
                "10_zone",
                80,
                {
                    0.08: 10,
                    0.16: 9,
                    0.24: 8,
                    0.32: 7,
                    0.4: 6,
                    0.48: 5,
                    0.56: 4,
                    0.64: 3,
                    0.72: 2,
                    0.8: 1,
                },
            ),
            ("WA_field", 80, {0.08: 6, 0.16: 5, 0.32: 4, 0.48: 3, 0.64: 2, 0.8: 1}),
            ("IFAA_field", 50, {0.1: 5, 0.3: 4, 0.5: 3}),
            ("Beiter_hit_miss", 6, {0.06: 1}),
            (
                "Worcester",
                (16, "inch"),
                {0.08128: 5, 0.16256: 4, 0.24384: 3, 0.32512: 2, 0.4064: 1},
            ),
            (
                "10_zone_6_ring",
                80,
                {0.08: 10, 0.16: 9, 0.24: 8, 0.32: 7, 0.4: 6, 0.48: 5},
            ),
            (
                "10_zone_5_ring_compound",
                40,
                {0.02: 10, 0.08: 9, 0.12: 8, 0.16: 7, 0.2: 6},
            ),
        ],
    )
    def test_get_face_spec(self, scoring_system, diam, expected_spec) -> None:
        ...
```

Nope, still nasty. Up to you, I just found it really hard to grok how many parameterisations there were and what was going into each invidudal test case while I was writing/debugging some of this in this format, but it is more technically correct.

Reply:
Or, at the cost of a lot of lines, I can put extra trailing commas in and force everything to multi line:

```python
    @pytest.mark.parametrize(
        "scoring_system, diam, expected_spec",
        [
            (
                "5_zone",
                122,
                {
                    0.244: 9,
                    0.488: 7,
                    0.732: 5,
                    0.976: 3,
                    1.22: 1,
                },
            ),
            (
                "10_zone",
                80,
                {
                    0.08: 10,
                    0.16: 9,
                    0.24: 8,
                    0.32: 7,
                    0.4: 6,
                    0.48: 5,
                    0.56: 4,
                    0.64: 3,
                    0.72: 2,
                    0.8: 1,
                },
            ),
            (
                "WA_field",
                80,
                {
                    0.08: 6,
                    0.16: 5,
                    0.32: 4,
                    0.48: 3,
                    0.64: 2,
                    0.8: 1,
                },
            ),
            (
                "IFAA_field",
                50,
                {
                    0.1: 5,
                    0.3: 4,
                    0.5: 3,
                },
            ),
            (
                "Beiter_hit_miss",
                6,
                {
                    0.06: 1,
                },
            ),
            (
                "Worcester",
                (16, "inch"),
                {
                    0.08128: 5,
                    0.16256: 4,
                    0.24384: 3,
                    0.32512: 2,
                    0.4064: 1,
                },
            ),
            (
                "10_zone_6_ring",
                80,
                {
                    0.08: 10,
                    0.16: 9,
                    0.24: 8,
                    0.32: 7,
                    0.4: 6,
                    0.48: 5,
                },
            ),
            (
                "10_zone_5_ring_compound",
                40,
                {0.02: 10, 0.08: 9, 0.12: 8, 0.16: 7, 0.2: 6},
            ),
        ],
    )
    def test_get_face_spec(self, scoring_system, diam, expected_spec) -> None:
        ...
```

Reply:
Not perfect, but I think the multiline is the most readable, and space not too much of a concern in tests.